File: MyVideoController/app/src/main/java/com/jwpeng/myvideocontroller/MainActivity.java
package com.jwpeng.myvideocontroller;

import androidx.appcompat.app.AppCompatActivity;

import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.widget.MediaController;
import android.widget.Toast;
import android.widget.VideoView;

public class MainActivity extends AppCompatActivity {

    VideoView videoView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate ( savedInstanceState );
        setContentView ( R.layout.activity_main );

        initView();
        initEnvent();
    }

    private void initEnvent() {
        String videouri = Utils.videoUri;
        Uri uri = Uri.parse ( videouri );
        // 设置视频控制器
        videoView.setMediaController ( new MediaController ( this ) );
        videoView.setOnCompletionListener ( new MyPlayerOnCompletionListener() );
        videoView.setVideoURI ( uri );
        videoView.start ();
    }

    private void initView() {
        videoView = findViewById ( R.id.video );
    }

    class MyPlayerOnCompletionListener implements MediaPlayer.OnCompletionListener {
        @Override
        public void onCompletion(MediaPlayer mp) {
            Toast.makeText ( MainActivity.this, "播放完成", Toast.LENGTH_SHORT ).show ();
        }
    }
}

File: MyEmail/app/src/main/java/com/jwpeng/mygps/MainActivity.kt
package com.jwpeng.mygps


import android.annotation.SuppressLint
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    var CC = arrayOf("2431652649@qq.com")

    @SuppressLint("IntentReset")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        sendbutton.setOnClickListener { 
            val intent = Intent(Intent.ACTION_SEND)
            intent.setData(Uri.parse("mailto"))
            intent.setType("text/plain")
            intent.putExtra(Intent.EXTRA_EMAIL, arrayOf(send_email.text.toString()))
            intent.putExtra(Intent.EXTRA_TEXT,text1_email.text.toString())
            intent.putExtra(Intent.EXTRA_SUBJECT,title_email.text.toString())
            intent.putExtra(Intent.EXTRA_CC,CC)
            intent.putExtra(Intent.EXTRA_BCC,CC)
            startActivity(Intent.createChooser(intent,"发送Email...."))

        }
    }

}